ingestion/src/metadata/ingestion/source/dashboard/mode/client.py

```diff
@@ -75,7 +75,7 @@ def fetch_all_reports(self, workspace_name: str) -> Optional[list]:
             dict
         """
         all_reports = []
-        response_collections = self.client.get(f"/{workspace_name}/{COLLECTIONS}?filter=all")
+        response_collections = self.client.get(f"/{workspace_name}/{COLLECTIONS}?filter=custom")
```

Comment:
Hi @ulixius9 I have made this change from `all` to `custom`

To answer your inquiry, all API Workspace tokens in Mode will have admin privileges by default (also only admins are able to create them) and according to the [docs](https://mode.com/help/articles/workspace-api-tokens), it looks like granular permission control are not yet supported.

Personal tokens that support granular permssion control to Mode resources appears to be deprecated according to this [docs](https://mode.com/developer/api-reference/authentication/), and no new personal tokens are able to be created as of now. This will be fully deprecated by August.

Several points when using the new Workspace Token:
- At least in my case here, not specifying the filter parameter returns no collections. I have tried it with both Restricted and Unrestricted (default access) collections.
- Specifying `custom` as a filter returns all collections in the Workspace, except Personal collections -> I think this is the most appropriate approach.
- Specifying 'all' returns all collections including Personal collections.

However, there is no way that I could confirm whether the existing users that are still using the deprecated Personal tokens (with limited access, as opposed to the new Workspace token method that grants default admin access) are able to use the `filter=custom` params without any errors, as I am also unable to create a new Personal token in Mode.

Perhaps should we hold off this PR until the usage of Personal Tokens are fully deprecated in August 2024?  @ulixius9

Reply:
Thanks @dixon2678 for this details explanation, IMO we can approach it this way -> we introduce a new field in the service connection form so that user can configure this filter i.e wether they want an `all`, `custom` or no filter at all


this way it is possible for users to ingest or skip personal collections and also possible for them to use the deprecated version if they still decide to use is, what do you think about this approach?